common/aws/provider/connection.go

```diff
@@ -55,6 +55,17 @@ var regionIsValid = func(region string) (isValid bool) {
 	return
 }
 
+var Ticker *time.Ticker
+
+func init() {
+	delay, err := time.ParseDuration(config.AWSTimeBetweenRequests())
```

Comment:
It would be better to fail in main and give a good error than to return early here and have the program crash later. 

Let's add some methods for other packages to manage this variable, but initialize it with a default.
I also think it is better to use variables describing what this object _does_ as opposed to what this object _is_, e.g.
```go
const DefaultRateLimit = time.XXX

var rateLimiter = time.NewTicker(DefaultRateLimit)

func ResetRateLimiter(rate time.Duration) {
    rateLimiter.Stop()
    rateLimiter = time.NewTicker(rate)
}

func StopRateLimiter() {
    rateLimiter.Stop()
}
```

Reply:
FWIW, these suggestions are essentially all hacks. The real problem is that each aws api call uses `getConnection` each time, which is totally unnecessary. The "good" solution here is to remove that from each api call, and then allow `main` to pass in a `time.Ticker` as the rate limiter when constructing the aws api provider. However, that's a lot more work for a tiny win.

Reply:
If we're giving the ticker a default and managing it from outside, I think I can take the `init()` function back out.